sources/models/src/modeled_types/shared.rs

```diff
@@ -396,28 +408,18 @@ mod test_dns_domain {
 
     #[test]
     fn valid_dns_domain() {
-        for ok in &[
-            "cluster.local",
-            "dev.eks",
-            "stage.eks",
-            "prod.eks",
-            "1-good.local",
-            &"a".repeat(63),
-            &format!("a{}", ".a".repeat(126)),
```

Comment:
Do these no longer pass as valid domains?  It seems like "1-good.local" and "a.a.a" should, though maybe not "aaa"...

Reply:
Yes. Those will pass. Obviously invalid ones like `&format!("a{}", ".a".repeat(500))`, `&"a".repeat(500)`, and `_b_a_d_.com` will also pass. The `url::Host::parse()` checks are not as strict as what I had earlier. I think that's fine though. In this case I think it's better to have a standard library do the checks instead of custom code. Especially since the kubelet doesn't even do any checks on this. Since just about everything passes, I removed some of the test cases and used it more as documentation for what likely uses would look like.